### ISSUE TITLE
[Bug]: write_csv fails with filename provided as string

### ISSUE THREAD

**Issue by tarheels100** [spectrochempy/spectrochempy#706]
### What happened?

I tried saving a dataset using `data.write_csv(outpath)` where `outpath` is a string.  I received the following error:
```
 ERROR | AttributeError: 'str' object has no attribute 'exists'
```

When debugging, I got the following trace:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\aspencer\Anaconda3\envs\spectrochempy\lib\site-packages\spectrochempy\core\writers\write_csv.py", line 59, in write_csv
    return exporter(*args, **kwargs)
  File "C:\Users\aspencer\Anaconda3\envs\spectrochempy\lib\site-packages\spectrochempy\core\writers\exporter.py", line 63, in __call__
    raise e
  File "C:\Users\aspencer\Anaconda3\envs\spectrochempy\lib\site-packages\spectrochempy\core\writers\exporter.py", line 52, in __call__
    filename = check_filename_to_save(self.object, *args, **kwargs)
  File "C:\Users\aspencer\Anaconda3\envs\spectrochempy\lib\site-packages\spectrochempy\utils\file.py", line 667, in check_filename_to_save
    if not filename or save_as or filename.exists():
AttributeError: 'str' object has no attribute 'exists'
```
This seems to indicate that the issue is with line 667 of `file.py`:
```python
    if not filename or save_as or filename.exists():
```
This throws an error since filename is a string, which has no method `exists`.  I was able to work around the issue by providing the path as a pathlib.Path object instead, like this:
```python
data.write_csv(pathlib.Path(outpath))
```

### What did you expect to happen?

I expected the data to be saved at the path provided since this usage is consistent with the documentation.

### Minimal Complete Verifiable Example

```python
import spectrochempy as scp

# Need to add in your own file paths here.
datapath = r"C:\somewhere\file.spa"
outpath = r"C:\somewhere\file.csv"
data = scp.read(datapath)
data.write_csv(outpath)
```

### Environment

(spectrochempy) W:\Users\ASpencer\Python\IRspec>C:/Users/aspencer/Anaconda3/envs/spectrochempy/python.exe w:/Users/ASpencer/Python/IRspec/scptest.py
C:\Users\aspencer\Anaconda3\envs\spectrochempy\lib\site-packages\spectrochempy\application\application.py:35: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  from pkg_resources import DistributionNotFound, get_distribution
SpectroChemPy's API - v.0.6.7
© Copyright 2014-2023 - A.Travert & C.Fernandez @ LCS

INSTALLED VERSIONS
------------------
commit: None
python: 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:34:57) [MSC v.1936 64 bit (AMD64)]
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 140 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: ('en_US', 'ISO8859-1')

 ERROR | FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\aspencer\\Anaconda3\\envs\\spectrochempy\\lib\\site-packages\\requirements\\requirements_dev.txt'

**Comment by fernandezc** [comment]
Thanks you @tarheels100 for the report. I'll fix this issue for the next release.

**Comment by fernandezc** [comment]
> Thanks you @tarheels100 for the report. I'll fix this issue for the next release.

Hi,
I have corrected the bug, so the writing should work as expected with a filename provided as string. It is already available in the latest "dev" version and will be released soon in version 0.6.8.
